types/assert-plus/index.d.ts

```diff
@@ -7,103 +7,103 @@
 
-export function optionalFinite(options: number | undefined, message ?: string): void;
+export function optionalFinite(options: number | undefined, message ?: string): asserts options is number | undefined;
 
-export function optionalObject(options: any, message ?: string): void;
+export function optionalObject(options: object | undefined, message ?: string): asserts options is object | undefined;
 
-export function optionalString(options: string | undefined, message ?: string): void;
+export function optionalString(options: string | undefined, message ?: string): asserts options is string | undefined;
 
-export function optionalStream(options: Stream | undefined, message ?: string): void;
+export function optionalStream(options: Stream | undefined, message ?: string): asserts options is Stream | undefined;
 
-export function optionalDate(options: Date | undefined, message ?: string): void;
+export function optionalDate(options: Date | undefined, message ?: string): asserts options is Date | undefined;
 
-export function optionalRegexp(options: RegExp | undefined, message ?: string): void;
+export function optionalRegexp(options: RegExp | undefined, message ?: string): asserts options is RegExp | undefined;
 
-export function optionalUuid(options: string | undefined, message ?: string): void;
+export function optionalUuid(options: string | undefined, message ?: string): asserts options is string | undefined;
 
-export function optionalArrayOfArray(arr: any[][] | undefined, message ?: string): void;
+export function optionalArrayOfArray(arr: any[][] | undefined, message ?: string): asserts arr is any[][] | undefined;
 
-export function optionalArrayOfBool(arr: boolean[] | undefined, message ?: string): void;
+export function optionalArrayOfBool(arr: boolean[] | undefined, message ?: string): asserts arr is boolean[] | undefined;
 
-export function optionalArrayOfBuffer(arr: Buffer[] | undefined, message ?: string): void;
+export function optionalArrayOfBuffer(arr: Buffer[] | undefined, message ?: string): asserts arr is Buffer[] | undefined;
 
-export function optionalArrayOfFunc(arr: any[] | undefined, message ?: string): void;
+export function optionalArrayOfFunc(arr: Function[] | undefined, message ?: string): asserts arr is Function[] | undefined;
 
-export function optionalArrayOfNumber(arr: number[] | undefined, message ?: string): void;
+export function optionalArrayOfNumber(arr: number[] | undefined, message ?: string): asserts arr is number[] | undefined;
 
-export function optionalArrayOfFinite(arr: number[] | undefined, message ?: string): void;
+export function optionalArrayOfFinite(arr: number[] | undefined, message ?: string): asserts arr is number[] | undefined;
 
-export function optionalArrayOfObject(arr: any[] | undefined, message ?: string): void;
+export function optionalArrayOfObject(arr: object[] | undefined, message ?: string): asserts arr is object[] | undefined;
```

Comment:
This code should also fixed like example above with `object`. Please, add test for it.